### PR TITLE
Add end function call to lookup processes

### PR DESCRIPTION
**Pull request by dianashk**
Add stream-end handling in order to shut down local pip child processes

Refactored the resolvers to be classes with `lookup` and `end` functions. Invoke `end` function in the parallel-stream stream-end-handler.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "pelias-wof-admin-lookup",
-  "version": "1.1.2",
```
Comment (orangejulius) [inline package.json]: Minor tip: when incrementing version numbers, if you do it in a separate commit then anyone scanning a list of commits, whether on their local machine, on github, or in emails like the ones Greenkeeper sends us about what's changed in a module we require, will easily be able to see where each version change happened.

**Comment by orangejulius** [comment]
The code looks great, but I couldn't get the process to end after the import :/ I saw a link to a little script that will list all the things keeping your node process alive the other day, but I can't find it now.

**Comment by orangejulius** [comment]
Found it, and here's what i get:

```
2016-02-25T19:15:52.475Z - verbose: [openaddresses] number of invalid records skipped: 2058
[WTF Node?] open handles:
- Timers:
  - (10000 ~ 10 s) wrapper @ /home/julian/repos/pelias/openaddresses/node_modules/pelias-dbclient/src/stats.js:17
  - (120000 ~ 2 min)  @ /home/julian/repos/pelias/openaddresses/node_modules/pelias-dbclient/node_modules/elasticsearch/src/lib/transport.js:318
- Others:
  - ChildProcess
  - Pipe
  - ChildProcess
  - Pipe
  - ChildProcess
  - Pipe
  - ChildProcess
  - Pipe
  - ChildProcess
  - Pipe
  - ChildProcess
  - Pipe
  - ChildProcess
  - Pipe
  - ChildProcess
  - Pipe
  - ChildProcess
  - Pipe
  - ChildProcess
  - Pipe
2016-02-25T19:15:52.789Z - verbose: [dbclient]  paused=false, transient=0, current_length=0, indexed=173332, batch_ok=347, batch_retries=0, failed_records=0, address=173332, persec=832
2016-02-25T19:15:52.790Z - verbose: [dbclient]  paused=false, transient=0, current_length=0, indexed=173332, batch_ok=347, batch_retries=0, failed_records=0, address=173332, persec=832
```

**Comment by dianashk** [comment]
Figured out the issue. The `end` function was being called without `this` set to anything, because... javascript. Added a test assertion.

**Comment by dianashk** [comment]
```
$ npm publish
+ pelias-wof-admin-lookup@1.1.5
```

**Comment by orangejulius** [comment]
Excellent! :+1:
